executor/join.go

```diff
@@ -550,7 +556,11 @@ func (e *HashJoinExec) fetchInnerAndBuildHashTable(ctx context.Context) {
 	// innerResultCh transfer inner result chunk from inner fetch to build hash table.
 	innerResultCh := make(chan *chunk.Chunk, e.concurrency)
 	doneCh := make(chan struct{})
-	go util.WithRecovery(func() { e.fetchInnerRows(ctx, innerResultCh, doneCh) }, nil)
+	e.fetchInnerDone.Add(1)
+	go util.WithRecovery(
+		func() { e.fetchInnerRows(ctx, innerResultCh, doneCh) },
+		func(r interface{}) { e.finishFetchInnerRows(innerResultCh) },
```

Comment:
only `finishFetchInnerRows()` when the `fetchInnerRows()` goroutine is paniced?

Reply:
not~ it alway call finishFetchInnerRows but with `r ==nil`

Reply:
😂 I was mislead by the `r interface{}` parameter in the function call.

Reply:
can we move the cleaning code into that function, and only do the recover work in this function?